File: Iniciante/1079 - Médias Ponderadas/Python/1079.py
WEIGHT_A = 2.0
WEIGHT_B = 3.0
WEIGHT_C = 5.0
TOTAL_WEIGHT = WEIGHT_A + WEIGHT_B + WEIGHT_C


def main():
    NUMBER_OF_NOTES = 3

    n = int(input())
    for _ in range(n):
        notes = list(map(float, input().split()))
        average = calculate_average(notes)
        print(f"{average:.1f}")


def calculate_average(notes):
    return (
        (notes[0] * WEIGHT_A) + (notes[1] * WEIGHT_B) + (notes[2] * WEIGHT_C)
    ) / TOTAL_WEIGHT


if __name__ == "__main__":
    main()

File: Iniciante/1079 - Médias Ponderadas/TypeScript/1079.ts
import * as fs from "fs";

const WEIGHT_A: number = 2.0;
const WEIGHT_B: number = 3.0;
const WEIGHT_C: number = 5.0;
const TOTAL_WEIGHT: number = WEIGHT_A + WEIGHT_B + WEIGHT_C;

function main(): void {
  const input: string[] = fs.readFileSync(0, "utf8").trim().split(/\s+/);

  const NUMBER_OF_NOTES: number = 3;

  let index: number = 0;
  const n: number = parseInt(input[index++]);

  for (let i = 0; i < n; i++) {
    const notes: number[] = [];
    for (let j = 0; j < NUMBER_OF_NOTES; j++) {
      notes.push(parseFloat(input[index++]));
    }
    const average: number = calculateAverage(notes);
    console.log(average.toFixed(1));
  }
}

function calculateAverage(notes: number[]): number {
  return (
    (notes[0] * WEIGHT_A + notes[1] * WEIGHT_B + notes[2] * WEIGHT_C) /
    TOTAL_WEIGHT
  );
}


main();
